gapic-generator/lib/gapic/presenters/method_rest_presenter.rb

```diff
@@ -0,0 +1,186 @@
+# frozen_string_literal: true
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "gapic/uri_template"
+
+module Gapic
+  module Presenters
+    ##
+    # A presenter for rpc methods (REST submethods)
+    #
+    class MethodRestPresenter
+      def initialize main_method
+        @main_method = main_method
+        @proto_method = main_method.method
+      end
+
+      ##
+      # @return [Boolean] Whether a http verb is present for this method
+      #
+      def verb?
+        !verb.nil?
+      end
+
+      ##
+      # @return [Symbol] a http verb for this method
+      #
+      def verb
+        return nil if @proto_method.http.nil?
+
+        method = {
+          get:    @proto_method.http.get,
+          post:   @proto_method.http.post,
+          put:    @proto_method.http.put,
+          patch:  @proto_method.http.patch,
+          delete: @proto_method.http.delete
+        }.find { |_, value| !value.empty? }
+
+        method[0] unless method.nil?
+      end
+
+      ##
+      # @return [Boolean] Whether a method path is present and non-empty
+      #
+      def path?
+        !path.empty?
+      end
+
+      ##
+      # @return [String] A method path or nil if not present
```

Comment:
This seems to return the empty string, not nil, if not present?

Reply:
:+1: